src/LibraryManager.Vsix/Json/Completion/LibraryIdCompletionEntry.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.Imaging.Interop;
+using Microsoft.VisualStudio.Language.Intellisense;
+using Microsoft.Web.Editor.Completion;
+using Microsoft.VisualStudio.Text;
+using System.Diagnostics;
+
+namespace Microsoft.Web.LibraryManager.Vsix
+{
+    internal class LibraryIdCompletionEntry : SimpleCompletionEntry
+    {
+        internal int Major { get; private set; }
+
+        internal int Minor { get; private set; }
+
+        internal int Patch { get; private set; }
+
+        internal bool IsValid { get; private set; }
+
+        public LibraryIdCompletionEntry(string displayText, string insertionText, ImageMoniker moniker, IIntellisenseSession session)
+            : base(displayText, insertionText, moniker, session)
+        {
+        }
+
+        public LibraryIdCompletionEntry(string displayText, string insertionText, string description, ImageMoniker moniker, ITrackingSpan span, IIntellisenseSession session, bool versionSpecific = false)
+            : base(displayText, insertionText, description, moniker, span, session)
+        {
+            if (versionSpecific)
+            {
+                SetVersion(insertionText);
+            }
+        }
+
+        protected override int InternalCompareTo(CompletionEntry other)
+        {
+            if (other == null)
+            {
+                return 1;
+            }
+
+            LibraryIdCompletionEntry otherEntry = other as LibraryIdCompletionEntry;
```

Comment:
>otherEntry [](start = 37, length = 10)

You probably want to null check this.

Reply:
Yes. I think it is better to null check after that.

---
In reply to: [198551833](https://github.com/aspnet/LibraryManager/pull/258#discussion_r198551833) [](ancestors = 198551833)